tests/unit/conftest.py

```diff
@@ -0,0 +1,49 @@
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+import os
+from unittest.mock import PropertyMock
+
+import pytest
+from ops import JujuVersion
+from pytest_mock import MockerFixture
+
+
+@pytest.fixture(autouse=True)
+def juju_has_secrets(mocker: MockerFixture):
+    """This fixture will force the usage of secrets whenever run on Juju 3.x.
+
+    NOTE: This is needed, as normally JujuVersion is set to 0.0.0 in tests
+    (i.e. not the real juju version)
+    """
+    if juju_version := os.environ.get("LIBJUJU"):
+        juju_version = juju_version[1:].split(".")[0]
+    else:
+        juju_version = "3"
```

Comment:
This should be changed to fixture params, but that doesn't work with `unittest.TestCase` suites. I propose refactoring that after the sprint.

Reply:
Please create a backlog ticket for the refactoring (to do not forget in a rush).